drivers/fuel_gauge/ltc2959/ltc2959.c

```diff
@@ -0,0 +1,544 @@
+			break;
+
+		default:
+			LOG_ERR("Unsupported GPIO analog mode: 0x%x", gpio_mode);
+			return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int ltc2959_get_gpio_threshold_uv(const struct device* dev, bool high, int32_t* value_uv)
+{
+	uint8_t reg = high ? LTC2959_REG_GPIO_THRESH_HIGH_MSB : LTC2959_REG_GPIO_THRESH_LOW_MSB;
+	const struct ltc2959_config* cfg = dev->config;
+	uint8_t ctrl;
+	int ret = i2c_reg_read_byte_dt(&cfg->i2c, LTC2959_REG_ADC_CONTROL, &ctrl);
+
+	if (ret < 0) {
+		LOG_ERR("NO CTRL: %i", ret);
+		return ret;
+	}
+
+	uint16_t raw;
+	ret = ltc2959_read16(dev, reg, &raw);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	int16_t raw_signed = (int16_t) raw;
+	uint8_t gpio_mode = ctrl & LTC2959_CTRL_GPIO_MODE_MASK;
+
+	switch (gpio_mode) {
+		case LTC2959_GPIO_MODE_BIPOLAR:
+			*value_uv = ((int64_t) raw_signed * LTC2959_GPIO_BIPOLAR_UV_SF) >> 15;
+			break;
+
+		case LTC2959_GPIO_MODE_UNIPOLAR:
+			*value_uv = ((uint32_t) raw * LTC2959_GPIO_UNIPOLAR_UV_SF) >> 15;
+			break;
+
+		default:
+			LOG_ERR("Unsupported GPIO mode: 0x%x", gpio_mode);
+			return -ENOTSUP;
+	}
+	return 0;
+}
+
+static int ltc2959_set_gpio_threshold_uv(const struct device* dev, bool high, int32_t value_uv)
+{
+	uint8_t reg = high ? LTC2959_REG_GPIO_THRESH_HIGH_MSB : LTC2959_REG_GPIO_THRESH_LOW_MSB;
+
+	const struct ltc2959_config* cfg = dev->config;
+
+	uint8_t ctrl;
+	int ret = i2c_reg_read_byte_dt(&cfg->i2c, LTC2959_REG_ADC_CONTROL, &ctrl);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	uint8_t gpio_mode = ctrl & LTC2959_CTRL_GPIO_MODE_MASK;
+
+	switch (gpio_mode) {
+		case LTC2959_GPIO_MODE_BIPOLAR: {
+			int16_t raw = (int16_t) (((int64_t) value_uv * 32768) /
+			        LTC2959_GPIO_BIPOLAR_UV_SF);
+			uint8_t buf[2] = {raw >> 8, raw & 0xFF};
+			return i2c_burst_write_dt(&cfg->i2c, reg, buf, sizeof(buf));
+		}
+		case LTC2959_GPIO_MODE_UNIPOLAR: {
+			uint16_t raw = (uint16_t) (((int64_t) value_uv * 32768) /
+			        LTC2959_GPIO_UNIPOLAR_UV_SF);
+			uint8_t buf[2] = {raw >> 8, raw & 0xFF};
+			return i2c_burst_write_dt(&cfg->i2c, reg, buf, sizeof(buf));
+		}
+		default:
+			break;
+	}
+
+	LOG_ERR("Unsupported GPIO mode: 0x%02x", gpio_mode);
+	return -ENOTSUP;
+}
+
+static int ltc2959_get_voltage_threshold_uv(const struct device* dev, bool high, uint32_t* value)
+{
+	uint8_t reg = high ? LTC2959_REG_VOLT_THRESH_HIGH_MSB : LTC2959_REG_VOLT_THRESH_LOW_MSB;
+	uint16_t raw;
+	int ret = ltc2959_read16(dev, reg, &raw);
+
+	if (ret < 0) {
+		LOG_ERR("Failed to get voltage threshold: %i", ret);
+		return ret;
+	}
+
+	*value = ((uint64_t) raw * LTC2959_VOLT_THRESH_UV_SCALAR) >> 15;
+
+	return 0;
+}
+
+static int ltc2959_set_voltage_threshold_uv(const struct device* dev, bool high, uint32_t value)
+{
+	uint8_t reg = high ? LTC2959_REG_VOLT_THRESH_HIGH_MSB : LTC2959_REG_VOLT_THRESH_LOW_MSB;
+	const uint16_t raw = (((uint64_t) value << 15) / LTC2959_VOLT_THRESH_UV_SCALAR);
+
+	uint8_t buf[2] = {raw >> 8, raw & 0xFF};
```

Comment:
Better use `sys_put_be16()` or from `byteorder.h`
```suggestion
	uint8_t buf[2];
	sys_put_be16(raw, buf);
```